deobfuscator/comm.js

```diff
@@ -201,29 +233,49 @@ var SandboxEval = function () {
     var sandbox = document.createElement('iframe');
     sandbox.id ='sandbox';
     sandbox.sandbox ='allow-scripts'; //sandboxing
-
+    sandbox.style.cssText = "width:1px;height:1px;display:none;visibility:hidden";
     var html = "\x3Cscript>(" + receiveAndPassBack.toString().replace(/"%ORIGIN%"/g, '"' + window.location.origin + '"') + ")();\x3C/script>";
     sandbox.src = 'data:text/html;charset=utf-8,' + encodeURI(html);
+    var sandboxLoaded = new Promise(function(resolve, reject) {
+        sandbox.onload = function() {
+            resolve("success");
+        }
+        sandbox.onerror = function(er) { reject(er);};
+    });
     document.body.appendChild(sandbox);
 
-    // register a local variable that we store the returned result.
-    var result;
-
-    function returnResult(evt) {
-        if(evt.success == 1) {
-            resolve(evt.data.result);
-        }
-        else if(evt.sucess == 0) {
-            reject(evt.data.error);
-        }
-        else {
-            reject(Error("Internal error."));
-        }
-    }
+    this.eval = function(code) {
+        return sandboxLoaded.catch(function(error){
+            ;
```

Comment:
I also think that it is messy, I'd appreciate if you could provide some advice on how I can improve it.